File: 3.x.x/crates-utopia/span/src/storage/memory.rs
use crate::span::Span;
use crate::storage::Storage;

/// Simple memory storage (PHP `Storage\Memory`).
#[derive(Debug, Default)]
pub struct Memory {
    span: parking_lot::Mutex<Option<Span>>,
}

impl Memory {
    pub fn new() -> Self {
        Self::default()
    }
}

impl Storage for Memory {
    fn get(&self) -> Option<Span> {
        self.span.lock().clone()
    }

    fn set(&self, span: Option<Span>) {
        *self.span.lock() = span;
    }
}

File: packages/span/src/Storage/Memory.php
<?php

declare(strict_types=1);

namespace Utopia\Span\Storage;

use Utopia\Span\Span;

/**
 * Simple memory storage for single-process PHP (FPM, CLI)
 */
class Memory implements Storage
{
    private ?Span $span = null;

    public function get(): ?Span
    {
        return $this->span;
    }

    public function set(?Span $span): void
    {
        $this->span = $span;
    }
}
